XMLDocParser extracts empty field names

**Describe the bug**

I am trying to run fscrawler (2.10-SNAPSHOT) and feed it into opensearch using docker-containers. 
I checked that the files are present and visible in the container.

Somehow only a single document gets added. After adding `--trace` to the command I noticed that an error gets produced when trying to convert a XML document:

`Invalid UTF-8 middle byte 0xcf`. This looks similar to https://github.com/dadoonet/fscrawler/issues/269 but this was fixed already.

**Job Settings**

```yml
---
name: "sample_files"
fs:
  url: "/tmp/es"
  update_rate: "15m"
  includes:
  - "*/*.doc"
  - "*/*.docx"
  - "*/*.pdf"
  - "*/*.xml"
  json_support: false
  filename_as_id: true
  add_filesize: true
  remove_deleted: true
  add_as_inner_object: false
  store_source: false
  index_content: true
  attributes_support: false
  raw_metadata: false
  xml_support: true
  index_folders: true
  lang_detect: false
  continue_on_error: false
  ocr:
    language: "eng"
    enabled: true
    pdf_strategy: "ocr_and_text"
  follow_symlinks: false
elasticsearch:
  nodes:
  - url: "https://opensearch-node1:9200"
  bulk_size: 100
  flush_interval: "5s"
  byte_size: "10mb"
  ssl_verification: false  # true by default
  username: "admin"
  password: "admin"
```

**Logs**

```
11:48:44,813 TRACE [f.p.e.c.f.t.XmlDocParser] Converting XML document
11:48:44,813 TRACE [f.p.e.c.f.t.XmlDocParser] Converting XML document
11:48:44,817 WARN  [f.p.e.c.f.FsParserAbstract] Error while crawling /tmp/es: com.fasterxml.jackson.core.JsonParseException: Invalid UTF-8 middle byte 0xcf (at char #0, byte #-1)
11:48:44,817 WARN  [f.p.e.c.f.FsParserAbstract] Full stacktrace
java.lang.RuntimeException: com.fasterxml.jackson.core.JsonParseException: Invalid UTF-8 middle byte 0xcf (at char #0, byte #-1)
	at fr.pilato.elasticsearch.crawler.fs.tika.XmlDocParser.asMap(XmlDocParser.java:82) ~[fscrawler-tika-2.10-SNAPSHOT.jar:?]
	at fr.pilato.elasticsearch.crawler.fs.tika.XmlDocParser.generateMap(XmlDocParser.java:72) ~[fscrawler-tika-2.10-SNAPSHOT.jar:?]
	at fr.pilato.elasticsearch.crawler.fs.tika.XmlDocParser.generate(XmlDocParser.java:52) ~[fscrawler-tika-2.10-SNAPSHOT.jar:?]
	at fr.pilato.elasticsearch.crawler.fs.FsParserAbstract.indexFile(FsParserAbstract.java:486) ~[fscrawler-core-2.10-SNAPSHOT.jar:?]
	at fr.pilato.elasticsearch.crawler.fs.FsParserAbstract.addFilesRecursively(FsParserAbstract.java:277) ~[fscrawler-core-2.10-SNAPSHOT.jar:?]
	at fr.pilato.elasticsearch.crawler.fs.FsParserAbstract.addFilesRecursively(FsParserAbstract.java:304) ~[fscrawler-core-2.10-SNAPSHOT.jar:?]
	at fr.pilato.elasticsearch.crawler.fs.FsParserAbstract.run(FsParserAbstract.java:152) [fscrawler-core-2.10-SNAPSHOT.jar:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: com.fasterxml.jackson.core.JsonParseException: Invalid UTF-8 middle byte 0xcf (at char #0, byte #-1)
	at com.fasterxml.jackson.dataformat.xml.XmlFactory._initializeXmlReader(XmlFactory.java:734) ~[jackson-dataformat-xml-2.15.2.jar:2.15.2]
	at com.fasterxml.jackson.dataformat.xml.XmlFactory._createParser(XmlFactory.java:600) ~[jackson-dataformat-xml-2.15.2.jar:2.15.2]
	at com.fasterxml.jackson.dataformat.xml.XmlFactory._createParser(XmlFactory.java:30) ~[jackson-dataformat-xml-2.15.2.jar:2.15.2]
	at com.fasterxml.jackson.core.JsonFactory.createParser(JsonFactory.java:1143) ~[jackson-core-2.15.2.jar:2.15.2]
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3809) ~[jackson-databind-2.15.2.jar:2.15.2]
	at fr.pilato.elasticsearch.crawler.fs.tika.XmlDocParser.asMap(XmlDocParser.java:80) ~[fscrawler-tika-2.10-SNAPSHOT.jar:?]
	... 7 more
Caused by: com.ctc.wstx.exc.WstxIOException: Invalid UTF-8 middle byte 0xcf (at char #0, byte #-1)
	at com.ctc.wstx.sr.StreamScanner.constructFromIOE(StreamScanner.java:653) ~[woodstox-core-6.5.1.jar:6.5.1]
	at com.ctc.wstx.sr.StreamScanner.loadMore(StreamScanner.java:1017) ~[woodstox-core-6.5.1.jar:6.5.1]
	at com.ctc.wstx.sr.StreamScanner.getNext(StreamScanner.java:770) ~[woodstox-core-6.5.1.jar:6.5.1]
	at com.ctc.wstx.sr.BasicStreamReader.nextFromProlog(BasicStreamReader.java:2096) ~[woodstox-core-6.5.1.jar:6.5.1]
	at com.ctc.wstx.sr.BasicStreamReader.next(BasicStreamReader.java:1180) ~[woodstox-core-6.5.1.jar:6.5.1]
	at com.fasterxml.jackson.dataformat.xml.XmlFactory._initializeXmlReader(XmlFactory.java:729) ~[jackson-dataformat-xml-2.15.2.jar:2.15.2]
	at com.fasterxml.jackson.dataformat.xml.XmlFactory._createParser(XmlFactory.java:600) ~[jackson-dataformat-xml-2.15.2.jar:2.15.2]
	at com.fasterxml.jackson.dataformat.xml.XmlFactory._createParser(XmlFactory.java:30) ~[jackson-dataformat-xml-2.15.2.jar:2.15.2]
	at com.fasterxml.jackson.core.JsonFactory.createParser(JsonFactory.java:1143) ~[jackson-core-2.15.2.jar:2.15.2]
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3809) ~[jackson-databind-2.15.2.jar:2.15.2]
	at fr.pilato.elasticsearch.crawler.fs.tika.XmlDocParser.asMap(XmlDocParser.java:80) ~[fscrawler-tika-2.10-SNAPSHOT.jar:?]
	... 7 more
Caused by: java.io.CharConversionException: Invalid UTF-8 middle byte 0xcf (at char #0, byte #-1)
	at com.ctc.wstx.io.UTF8Reader.reportInvalidOther(UTF8Reader.java:318) ~[woodstox-core-6.5.1.jar:6.5.1]
	at com.ctc.wstx.io.UTF8Reader.read(UTF8Reader.java:208) ~[woodstox-core-6.5.1.jar:6.5.1]
	at com.ctc.wstx.io.ReaderSource.readInto(ReaderSource.java:88) ~[woodstox-core-6.5.1.jar:6.5.1]
	at com.ctc.wstx.io.BranchingReaderSource.readInto(BranchingReaderSource.java:57) ~[woodstox-core-6.5.1.jar:6.5.1]
	at com.ctc.wstx.sr.StreamScanner.loadMore(StreamScanner.java:1011) ~[woodstox-core-6.5.1.jar:6.5.1]
	at com.ctc.wstx.sr.StreamScanner.getNext(StreamScanner.java:770) ~[woodstox-core-6.5.1.jar:6.5.1]
	at com.ctc.wstx.sr.BasicStreamReader.nextFromProlog(BasicStreamReader.java:2096) ~[woodstox-core-6.5.1.jar:6.5.1]
	at com.ctc.wstx.sr.BasicStreamReader.next(BasicStreamReader.java:1180) ~[woodstox-core-6.5.1.jar:6.5.1]
	at com.fasterxml.jackson.dataformat.xml.XmlFactory._initializeXmlReader(XmlFactory.java:729) ~[jackson-dataformat-xml-2.15.2.jar:2.15.2]
	at com.fasterxml.jackson.dataformat.xml.XmlFactory._createParser(XmlFactory.java:600) ~[jackson-dataformat-xml-2.15.2.jar:2.15.2]
	at com.fasterxml.jackson.dataformat.xml.XmlFactory._createParser(XmlFactory.java:30) ~[jackson-dataformat-xml-2.15.2.jar:2.15.2]
	at com.fasterxml.jackson.core.JsonFactory.createParser(JsonFactory.java:1143) ~[jackson-core-2.15.2.jar:2.15.2]
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3809) ~[jackson-databind-2.15.2.jar:2.15.2]
	at fr.pilato.elasticsearch.crawler.fs.tika.XmlDocParser.asMap(XmlDocParser.java:80) ~[fscrawler-tika-2.10-SNAPSHOT.jar:?]
	... 7 more
```

**Expected behavior**

- The file gets parsed correctly.
- That the other files that did not have issues are parsed.

**Versions:**
(I am using docker though)
 - OS: MacOS
 - Version: 13.6

--- a/integration-tests/src/test/java/fr/pilato/elasticsearch/crawler/fs/test/integration/elasticsearch/FsCrawlerTestXmlSupportIT.java
+++ b/integration-tests/src/test/java/fr/pilato/elasticsearch/crawler/fs/test/integration/elasticsearch/FsCrawlerTestXmlSupportIT.java
@@ -57,4 +57,27 @@ public class FsCrawlerTestXmlSupportIT extends AbstractFsCrawlerITCase {
             logger.info("{}", hit.getSource());
         }
     }
+
+    /**
+     * Test case for issue #1753: <a href="https://github.com/dadoonet/fscrawler/issues/1753">https://github.com/dadoonet/fscrawler/issues/1753</a> :
+     * invalid json generated from XML
+     */
+    @Test
+    public void test_xml_not_readable() throws Exception {
+        Fs fs = startCrawlerDefinition()
+                .setXmlSupport(true)
+                .build();
+        crawler = startCrawler(getCrawlerName(), fs, endCrawlerDefinition(getCrawlerName()), null);
+        ESSearchResponse response = countTestHelper(new ESSearchRequest().withIndex(getCrawlerName()), 1L, null);
+
+        countTestHelper(new ESSearchRequest()
+                .withIndex(getCrawlerName())
+                .withESQuery(new ESMatchQuery("Tag.$", "Content")),
+                1L, null);
+
+        logger.info("XML documents converted to:");
+        for (ESSearchHit hit : response.getHits()) {
+            logger.info("{}", hit.getSource());
+        }
+    }
 }

--- a/integration-tests/src/test/resources-binary/samples/test_xml_not_readable/sample.xml
+++ b/integration-tests/src/test/resources-binary/samples/test_xml_not_readable/sample.xml
@@ -1,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Task>
+ <Tag attr="false">Content</Tag>
+</Task>

--- a/tika/src/main/java/fr/pilato/elasticsearch/crawler/fs/tika/XmlDocParser.java
+++ b/tika/src/main/java/fr/pilato/elasticsearch/crawler/fs/tika/XmlDocParser.java
@@ -41,7 +41,7 @@ public class XmlDocParser {
     private static final ObjectMapper xmlMapper;
 
     static {
-        xmlMapper = new XmlMapper();
+        xmlMapper = XmlMapper.xmlBuilder().nameForTextElement("$").build();
     }
 
     public static String generate(InputStream inputStream) {

--- a/tika/src/test/java/fr/pilato/elasticsearch/crawler/fs/tika/DocParserTestCase.java
+++ b/tika/src/test/java/fr/pilato/elasticsearch/crawler/fs/tika/DocParserTestCase.java
@@ -25,6 +25,14 @@ import java.io.InputStream;
 
 public class DocParserTestCase extends AbstractFSCrawlerTestCase {
     InputStream getBinaryContent(String filename) {
-        return getClass().getResourceAsStream("/documents/" + filename);
+        return getBinaryContent("/documents", filename);
+    }
+
+    InputStream getBinaryContent(String root, String filename) {
+        if (root == null) {
+            // We try in the same classpath
+            return getClass().getResourceAsStream(filename);
+        }
+        return getClass().getResourceAsStream(root + "/" + filename);
     }
 }

--- a/tika/src/test/java/fr/pilato/elasticsearch/crawler/fs/tika/XmlDocParserTest.java
+++ b/tika/src/test/java/fr/pilato/elasticsearch/crawler/fs/tika/XmlDocParserTest.java
@@ -21,7 +21,6 @@ package fr.pilato.elasticsearch.crawler.fs.tika;
 
 import org.junit.Test;
 
-import java.io.IOException;
 import java.io.InputStream;
 
 import static org.hamcrest.MatcherAssert.assertThat;
@@ -30,7 +29,7 @@ import static org.hamcrest.Matchers.is;
 public class XmlDocParserTest extends DocParserTestCase {
 
     @Test
-    public void testXml() throws IOException {
+    public void testXml() {
         String doc = extractFromFile("issue-163.xml");
         assertThat(doc, is("{\"version\":\"1.0\",\"subscription-update\":{\"subscriptionid\":\"0\",\"requestid\":\"0\"," +
                 "\"last_push\":\"2016-06-03 06:21:34\",\"current_push\":\"2016-06-03 06:21:37\",\"exec\":\"0.002\"," +
@@ -38,9 +37,21 @@ public class XmlDocParserTest extends DocParserTestCase {
     }
 
     @Test
-    public void testXmlNestedObjects() throws IOException {
+    public void testXmlNestedObjects() {
         String doc = extractFromFile("issue-592.xml");
         assertThat(doc, is("{\"object\":[{\"id\":\"1\",\"name\":\"foo\"},{\"id\":\"2\",\"name\":\"bar\"}]}"));
+    }
+
+    @Test
+    public void testXmlNotReadable() {
+        String doc = extractFromFile(null, "issue-1753.xml");
+        assertThat(doc, is("{\"Tag\":{\"attr\":\"false\",\"$\":\"Content\"}}"));
+    }
+
+
+    private String extractFromFile(String root, String filename) {
+        InputStream data = getBinaryContent(root, filename);
+        return XmlDocParser.generate(data);
     }
 
     private String extractFromFile(String filename) {

--- a/tika/src/test/resources/fr/pilato/elasticsearch/crawler/fs/tika/issue-1753.xml
+++ b/tika/src/test/resources/fr/pilato/elasticsearch/crawler/fs/tika/issue-1753.xml
@@ -1,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Task>
+ <Tag attr="false">Content</Tag>
+</Task>

